cawg_identity/src/claim_aggregation/ica_signature_verifier.rs

```diff
@@ -69,59 +80,389 @@ impl SignatureVerifier for IcaSignatureVerifier {
+            if let Err(err) = self
+                .check_valid_until(&valid_until, maybe_tst_info.as_ref())
+                .await
+            {
+                // NOTE: We handle logging here because we want to signal at most one of the
+                // possible error conditions that are detectable in `check_valid_until`, BUT
+                // they are not fatal to the overall interpretation of the identity assertion.
+                //
+                // In the event that the status tracker is configured to proceed when possible,
+                // we log the error condition related to the signature and proceed.
+                ok = false;
+
+                log_current_item!(err.clone(), "IcaSignatureVerifier::check_signature")
+                    .validation_status("cawg.ica.valid_until.invalid")
+                    .failure(
+                        status_tracker,
+                        ValidationError::SignatureError(IcaValidationError::InvalidValidFromDate(
+                            err,
+                        )),
+                    )?;
+            }
+        }
+
+        // TO DO (CAI-7993): CAWG SDK should check ICA issuer revocation status.
+
+        // Verify that signer_payload is the same as credential issuer signed.
+        {
+            let subject = ica_credential.credential_subjects.first();
+
+            // The `DynamicAssertion` mechanism doesn't always populate the `alg` field when
+            // offering the partial claim for signature but some signers fill that in with a
+            // default value. Work around this by copying only the `alg` value from inside
+            // the VC.
+            let mut signer_payload = signer_payload.clone();
+            let new_ras = signer_payload
+                .referenced_assertions
+                .iter()
+                .zip(subject.c2pa_asset.referenced_assertions.iter())
+                .map(|(sp_ra, vc_ra)| {
+                    if sp_ra.alg().is_none() {
+                        HashedUri::new(sp_ra.url(), vc_ra.alg(), &sp_ra.hash())
+                    } else {
+                        sp_ra.clone()
+                    }
+                })
+                .collect();
+
+            signer_payload.referenced_assertions = new_ras;
+
+            if signer_payload != subject.c2pa_asset {
+                ok = false;
+
+                log_current_item!(
+                    "c2paAsset does not match signer_payload",
+                    "IcaSignatureVerifier::check_signature"
+                )
+                .validation_status("cawg.ica.signer_payload.mismatch")
+                .failure(
+                    status_tracker,
+                    ValidationError::SignatureError(IcaValidationError::SignerPayloadMismatch),
+                )?;
+            }
+        }
+
+        // TO DO (CAI-7994): CAWG SDK should inspect verifiedIdentities array.
+
+        if ok {
+            log_current_item!(
```

Comment:
So we get there only if all of the above is "ok". And specifically to log it is ok too here. That's a very long peice of code to rely on that variable for.